pkg/util/cloudinit/cloudconfig.go

```diff
@@ -139,9 +139,11 @@ func MarshalYAML(
 			// a string from the Content field.
 			content := secret.WriteFiles[in.WriteFiles[i].Path]
 			if content == "" {
-				if err := json.Unmarshal(
-					in.WriteFiles[i].Content, &content); err != nil {
-
+				inContent := in.WriteFiles[i].Content
+				if len(inContent) == 0 {
```

Comment:
Not a big deal but I think `if inContent == nil` is a little clearer here (or otherwise won't we assign the same thing below?)

Reply:
No.  `[]byte(`""`)` is not the same as `len(inContent) == 0`

Reply:
In the above case, the `inContent` is literally the JSON as it would be written in a JSON doc. It's "raw JSON" as it were.

Reply:
Therefore even `[]byte{}` would be invalid since it is not an "empty string" per the JSON rules.